File: linked_lists/linked_List.rb
class LinkedList

  attr_reader :head

  def initialize
    @head = nil
  end

  def push(value)
    @head = Node.new(value) if @head.nil?

    tail = find_tail
    tail.next = Node.new(value)
  end

  def find_tail(node = @head)
    return node unless node.next
    find_tail(node.next)
  end
end

class Node

  attr_reader :value
  attr_accessor :next

  def initialize(value)
    @value = value
    @next = nil
  end
end


require "pry"; binding.pry
""

File: linked_lists/linked_list.js
class LinkedList {
  constructor() {
    this.head = null
  }


  push(value) {
    if(this.head === null) {
      this.head = new Node(value)
    } else {
      var tail = this.findTail()

      tail.next = new Node(value)
    }
  }

  findTail(node = this.head) {
    while (node.next) {
      node = node.next
    }
    return node
  }
}


class Node {
  constructor(value) {
    this.value = value
    this.next = null
  }
}

ll = new LinkedList
ll.push('dog');
ll.push('cat');
ll.push('cats');
console.log(ll)